File: src/world/ColumnInfo.ts
import type { BiomeId } from './Biomes';

export interface ColumnInfo {
  /** Topmost fully solid gameplay block. */
  height: number;
  biome: BiomeId;
  surface: number;
  /**
   * Sub-block surface elevation in terrain voxels (0..SURFACE_STEPS-1).
   * 0 means the surface sits exactly on the block top (legacy behaviour).
   */
  step: number;
}

File: Assets/VYTHERA/Voxel/Storage/ColumnInfo.cs
using System;

namespace VYTHERA.Voxel.Storage
{
    [Serializable]
    public struct ColumnInfo
    {
        public int Height;
        public byte Biome;
        public int Surface;
        public int Step; // 0..3 (terrain sub-voxel resolution)
    }
}
